src/main/views/viewManager.ts

```diff
@@ -53,35 +60,204 @@ export enum LoadingScreenState {
+        ipcMain.on(UNREAD_RESULT, this.handleFaviconIsUnread);
+        ipcMain.handle(GET_VIEW_NAME, this.handleGetViewName);
     }
 
-    getServers = () => {
-        return Config.teams.concat();
+    init = () => {
+        this.getServers().forEach((server) => this.loadServer(server));
+        this.showInitial();
+    }
+
+    getView = (viewName: string) => {
+        return this.views.get(viewName);
+    }
+
+    getCurrentView = () => {
+        if (this.currentView) {
+            return this.views.get(this.currentView);
+        }
+
+        return undefined;
+    }
+
+    getViewByWebContentsId = (webContentsId: number) => {
+        return [...this.views.values()].find((view) => view.view.webContents.id === webContentsId);
+    }
+
+    showByName = (name: string) => {
+        log.debug('viewManager.showByName', name);
+
+        const newView = this.views.get(name);
+        if (newView) {
+            if (newView.isVisible) {
+                return;
+            }
+            if (this.currentView && this.currentView !== name) {
+                const previous = this.getCurrentView();
+                if (previous) {
+                    previous.hide();
+                }
+            }
+
+            this.currentView = name;
+            if (!newView.isErrored()) {
+                newView.show();
+                if (newView.needsLoadingScreen()) {
+                    this.showLoadingScreen();
+                }
+            }
+            MainWindow.get()?.webContents.send(SET_ACTIVE_VIEW, newView.tab.server.name, newView.tab.type);
+            ipcMain.emit(SET_ACTIVE_VIEW, true, newView.tab.server.name, newView.tab.type);
+            if (newView.isReady()) {
+                ipcMain.emit(UPDATE_LAST_ACTIVE, true, newView.tab.server.name, newView.tab.type);
+            } else {
+                log.warn(`couldn't show ${name}, not ready`);
+            }
+        } else {
+            log.warn(`Couldn't find a view with name: ${name}`);
+        }
+        modalManager.showModal();
+    }
+
+    focusCurrentView = () => {
+        if (modalManager.isModalDisplayed()) {
+            modalManager.focusCurrentModal();
+            return;
+        }
+
+        const view = this.getCurrentView();
+        if (view) {
+            view.focus();
+        }
+    }
+
+    reload = () => {
+        const currentView = this.getCurrentView();
+        if (currentView) {
+            this.showLoadingScreen();
+            currentView.reload();
+        }
+    }
+
+    sendToAllViews = (channel: string, ...args: unknown[]) => {
+        this.views.forEach((view) => {
+            if (!view.view.webContents.isDestroyed()) {
+                view.view.webContents.send(channel, ...args);
+            }
+        });
+    }
+
+    sendToFind = () => {
+        this.getCurrentView()?.openFind();
+    }
+
+    /**
+     * Deep linking
+     */
+
+    handleDeepLink = (url: string | URL) => {
+        // TODO: fix for new tabs
```

Comment:
do we have a ticket for this?

Reply:
Oh that's a super old TODO, I should just remove that cause I think it's working now anyways, and if it's not the new tabs are going to change it.